crates/re_log_types/src/lib.rs

```diff
@@ -75,6 +75,120 @@ macro_rules! impl_into_enum {
 
 // ----------------------------------------------------------------------------
 
+use arrow2_convert::{ArrowDeserialize, ArrowField, ArrowSerialize};
+
+#[derive(
+    Debug,
+    Clone,
+    Copy,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Hash,
+    ArrowField,
+    ArrowSerialize,
+    ArrowDeserialize,
+)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
+#[arrow_field(transparent)]
+pub struct TableId(re_tuid::Tuid);
+
+impl std::fmt::Display for TableId {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        self.0.fmt(f)
+    }
+}
+
+impl TableId {
+    pub const ZERO: Self = Self(re_tuid::Tuid::ZERO);
+
+    #[inline]
+    #[cfg(not(target_arch = "wasm32"))]
+    pub fn random() -> Self {
+        Self(re_tuid::Tuid::random())
+    }
+
+    /// Temporary utility while we transition to batching. See #1619.
+    #[doc(hidden)]
+    pub fn into_row_id(self) -> RowId {
+        RowId(self.0)
+    }
+}
+
+impl std::ops::Deref for TableId {
+    type Target = re_tuid::Tuid;
+
+    #[inline]
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl std::ops::DerefMut for TableId {
+    #[inline]
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.0
+    }
+}
+
+#[derive(
+    Debug,
+    Clone,
+    Copy,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Hash,
+    ArrowField,
+    ArrowSerialize,
+    ArrowDeserialize,
+)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
+#[arrow_field(transparent)]
+pub struct RowId(re_tuid::Tuid);
```

Comment:
I'd vote to put `TableID` into `table_id.rs` and `RowId` into `row_id.rs`. Makes them easier to find, and prevents the main `lib.rs` from growing forever

Reply:
Moved them closer to the types that they are IDs of